### PR TITLE
fix: Data-Analytics-Workspace - Several smaller adjustments to enable testing

### LINKED ISSUE #3598: [Failed pipeline] avm.ptn.data.private-analytical-workspace
Failed run: https://github.com/Azure/bicep-registry-modules/actions/runs/11427911716

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Disabled purge protection
- Renamed resources
- Added dynamic element to deployment names
- Upgraded used module versions & applied required updates
- Replaced Lock UDT with import

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.ptn.data.private-analytical-workspace](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.ptn.data.private-analytical-workspace.yml)     |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: avm/ptn/data/private-analytical-workspace/tests/common.tests.ps1 [hunk 1]
```diff
@@ -213,12 +213,12 @@ function Test-VerifyKeyVault($KeyVaultResourceGroupName, $KeyVaultName, $Tags, $
...
     $kv.EnableSoftDelete | Should -Be $EnableSoftDelete
     $kv.SoftDeleteRetentionInDays | Should -Be $RetentionInDays
-    $kv.EnablePurgeProtection | Should -Be $true
+    $kv.EnablePurgeProtection | Should -BeIn @($false, $null)
     $kv.PublicNetworkAccess | Should -Be $PublicNetworkAccess
     $kv.AccessPolicies | Should -BeNullOrEmpty
     $kv.NetworkAcls.DefaultAction | Should -Be 'Deny'
     $kv.NetworkAcls.Bypass | Should -Be 'None'
-    if ( $IpAddressRanges -eq $null ) { $kv.NetworkAcls.IpAddressRanges | Should -BeNullOrEmpty }
+    if ( $null -eq $IpAddressRanges ) { $kv.NetworkAcls.IpAddressRanges | Should -BeNullOrEmpty }
```
Comment (eriqua) [inline avm/ptn/data/private-analytical-workspace/tests/common.tests.ps1:221]: Why this change?

Comment (AlexanderSehr) [reply avm/ptn/data/private-analytical-workspace/tests/common.tests.ps1:221]: Because of 2 reasons:
1. It's a bad practice and other, less forgiving programming language can fail this check with a NullPointerException if you don't put the null-check to the left of the equation
2. The Script Analyzer highlighted the same with some very triggering yellow scribberly lines that I just had to address. Trust me, there was simply no other way

cc: @eriqua

Comment (eriqua) [reply avm/ptn/data/private-analytical-workspace/tests/common.tests.ps1:221]: Learn something new every day, thanks! 
https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-null?view=powershell-7.4#checking-for-null

**Comment by eriqua** [comment]
Lgtm 
@jbinko please have a look when you get some time 🙂

**Review by jbinko — APPROVED** [review]
Reviewed
